pkg/vf/rosetta_arm64.go

```diff
@@ -2,22 +2,32 @@ package vf
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/Code-Hex/vz/v3"
 	log "github.com/sirupsen/logrus"
 )
 
-func checkRosettaAvailability(install bool) error {
-	availability := vz.LinuxRosettaDirectoryShareAvailability()
+var (
+	checkRosettaDirectoryShareAvailability = vz.LinuxRosettaDirectoryShareAvailability
+	doInstallRosetta                       = vz.LinuxRosettaDirectoryShareInstallRosetta
+)
+
+func (dev *RosettaShare) checkRosettaAvailability() error {
+	availability := checkRosettaDirectoryShareAvailability()
 	switch availability {
 	case vz.LinuxRosettaAvailabilityNotSupported:
 		return fmt.Errorf("rosetta is not supported")
 	case vz.LinuxRosettaAvailabilityNotInstalled:
-		if !install {
+		if !dev.InstallRosetta {
 			return fmt.Errorf("rosetta is not installed")
 		}
 		log.Debugf("installing rosetta")
-		if err := vz.LinuxRosettaDirectoryShareInstallRosetta(); err != nil {
+		if err := doInstallRosetta(); err != nil {
+			if dev.IgnoreIfMissing && strings.Contains(err.Error(), fmt.Sprintf("VZErrorDomain Code=%d", vz.ErrorOperationCancelled)) {
+				log.Debugf("user cancelled rosetta installation but ignoreIfMissing option is true")
```

Comment:
Is there a way for (eg) podman to detect the rosetta installation failed? If not, we need to print something to stdout or stderr, which is not great, but will allow podman to know what happened.

Reply:
So, not sure what to do here and what's the suggested way to handle it. I just added a generic message to the stdout. Should I also print the error? I didn't print it on the stderr bc I think that if one listens to the stderr and receive something it assumes there was some kind of failure, but in this case we want to consciously skip it